From de6d848863396cb9dd11b2fceb758c9462bdcb14 Mon Sep 17 00:00:00 2001
From: glisse <glisse@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 26 Jun 2014 15:27:52 +0000
Subject: [PATCH] 2014-06-26  Marc Glisse  <marc.glisse@inria.fr>

	PR target/61503
	* config/i386/i386.md (x86_64_shrd, x86_shrd,
	ix86_rotr<dwi>3_doubleword): Replace ashiftrt with lshiftrt.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_9-branch@212036 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog           |    6 ++++++
 gcc/config/i386/i386.md |    8 ++++----
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index fde4557..e614c57 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,9 @@
+2014-06-26  Marc Glisse  <marc.glisse@inria.fr>
+
+	PR target/61503
+	* config/i386/i386.md (x86_64_shrd, x86_shrd,
+	ix86_rotr<dwi>3_doubleword): Replace ashiftrt with lshiftrt.
+
 2014-06-26  Martin Jambor  <mjambor@suse.cz>
 
 	Backport from mainline
diff --git a/gcc/config/i386/i386.md b/gcc/config/i386/i386.md
index e6bf58f..f677a7e 100644
--- a/gcc/config/i386/i386.md
+++ b/gcc/config/i386/i386.md
@@ -9598,7 +9598,7 @@
 
 (define_insn "x86_64_shrd"
   [(set (match_operand:DI 0 "nonimmediate_operand" "+r*m")
-        (ior:DI (ashiftrt:DI (match_dup 0)
+        (ior:DI (lshiftrt:DI (match_dup 0)
 		  (match_operand:QI 2 "nonmemory_operand" "Jc"))
 		(ashift:DI (match_operand:DI 1 "register_operand" "r")
 		  (minus:QI (const_int 64) (match_dup 2)))))
@@ -9614,7 +9614,7 @@
 
 (define_insn "x86_shrd"
   [(set (match_operand:SI 0 "nonimmediate_operand" "+r*m")
-        (ior:SI (ashiftrt:SI (match_dup 0)
+        (ior:SI (lshiftrt:SI (match_dup 0)
 		  (match_operand:QI 2 "nonmemory_operand" "Ic"))
 		(ashift:SI (match_operand:SI 1 "register_operand" "r")
 		  (minus:QI (const_int 32) (match_dup 2)))))
@@ -10066,13 +10066,13 @@
  [(set (match_dup 3) (match_dup 4))
   (parallel
    [(set (match_dup 4)
-	 (ior:DWIH (ashiftrt:DWIH (match_dup 4) (match_dup 2))
+	 (ior:DWIH (lshiftrt:DWIH (match_dup 4) (match_dup 2))
 		   (ashift:DWIH (match_dup 5)
 				(minus:QI (match_dup 6) (match_dup 2)))))
     (clobber (reg:CC FLAGS_REG))])
   (parallel
    [(set (match_dup 5)
-	 (ior:DWIH (ashiftrt:DWIH (match_dup 5) (match_dup 2))
+	 (ior:DWIH (lshiftrt:DWIH (match_dup 5) (match_dup 2))
 		   (ashift:DWIH (match_dup 3)
 				(minus:QI (match_dup 6) (match_dup 2)))))
     (clobber (reg:CC FLAGS_REG))])]
-- 
1.7.1

